pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -801,7 +808,13 @@ type PrometheusSpec struct {
 
 	// When true, the Prometheus compaction is disabled.
 	DisableCompaction bool `json:"disableCompaction,omitempty"`
-
+	// Configure the Go garbage collector GOGC parameter
+	// See: https://tip.golang.org/doc/gc-guide#GOGC
+	// Lowering this number increases CPU usage.
+	// It requires Prometheus >= v2.53.0.
+	// For lower Prometheus versions set `GOGC` env var.
```

Comment:
Joining @mviswanathsai here, we can make it work seamlessly for older Prometheus versions too via the env var. The downside is that changing the value requires a rollout, unlike setting the option in the config file.

Reply:
To clear my understanding did you mean we want to introduce a field to set environment variables for Prometheus not the [strategic merge patch way](https://prometheus-operator.dev/docs/operator/strategic-merge-patch/#how-to-inject-an-environment-variable-in-an-existing-container)?

Reply:
Since we introduce a field in the CRD to control the GOGC value, it makes sense to support it for all Prometheus versions if possible which is the case by setting the env var in the pod template. In this case yes the strategic merge patch option becomes deprecated.